lib/src/test/java/com/dynatrace/file/util/FilePollerTest.java

```diff
@@ -27,32 +83,43 @@ void filePollerConstructorThrowsOnPassingAFolder() throws IOException {
 
     assertThrows(
         IllegalArgumentException.class,
-        () -> {
-          new FilePoller(tempDir.toString());
-        });
+        () -> FilePollerFactory.getPollBased(tempDir.toString(), Duration.ofMillis(50)));
+
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> FilePollerFactory.getWatchServiceBased(tempDir.toString()));
   }
 
-  @Test
-  void filePollerUpdatesOnChange() throws IOException, InterruptedException {
-    final Path tempFile = Files.createTempFile("tempfile", ".txt");
+  void filePollerUpdatesOnChange(AbstractFilePoller poller) throws IOException {
```

Comment:
Maybe consider moving non-test methods below tests so it's not confusing. Initially I thought this was a test and it was missing the `@Test` attribute, but then I saw it was just a handy method.